### PR TITLE
feat: Enable semantic commit messages

### LINKED ISSUE #1987: Semantic Pull Requests via Conventional Commits
**What would you like to be added**:
Introduce a PR check that ensures commit messages follow https://www.conventionalcommits.org/en/v1.0.0/

**Why is this needed**:

Currently it's difficult to autogenerate changelogs and thus releases. We would like to have regular releases and need to have more automation around it.

**Describe the solution you'd like**

This could be done via: https://github.com/amannn/action-semantic-pull-request

**Additional context**
None

### PR DESCRIPTION
**Pull request by mrueg**
**What this PR does / why we need it**:

Adds documentation and a CI check for semantic commit messages.

**How does this change affect the cardinality of KSM**: *(increases, decreases or does not change cardinality)*
None
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1987

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -6,11 +6,40 @@ _As contributors and maintainers of this project, and in the interest of fosteri
...
 
-We have full documentation on how to get started contributing here: 
+We have full documentation on how to get started contributing here:
 
-<!---
-If your repo has certain guidelines for contribution, put them here ahead of the general k8s resources
--->
+### Semantic Commit Messages
+
+We use [semantic commit messages](https://www.conventionalcommits.org/en/v1.0.0/)  in this repository.
```
Comment (bavarianbidi) [inline CONTRIBUTING.md:13]:
```suggestion
We use [semantic commit messages](https://www.conventionalcommits.org/en/v1.0.0/) in this repository.
```

**Review by fpetkovski — COMMENTED** [review]
/approve

Worth keeping in mind that we don't strictly follow semver, so we might need to tweak the way we infer release versions (minor vs major).

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -6,11 +6,40 @@ _As contributors and maintainers of this project, and in the interest of fosteri
...
+
+You can also include a scope within parenthesis:
+
+`fix(scope): Prevent wrong calculation of storage`
+
+Here's a list of types that we use:
+
+| Type | Explanation |
+|---|---|
+| feat | A new feature |
```
Comment (CatherineF-dev) [inline CONTRIBUTING.md:29]: QQ: if a PR matches several types, does it support multiple types?

Comment (dgrisonnet) [reply CONTRIBUTING.md:29]: @mrueg could you perhaps answer this question in the doc?

**Review by dgrisonnet — COMMENTED** [review]
I am fine with trying this out.

/lgtm
/approve

**Pull request merged**